### PR TITLE
14487 fix contributing first steps

### LINKED ISSUE #14487: python3 runtests.py fails if python version == 3.11
**Bug Report**

Following the [contributor guide](https://github.com/python/mypy/blob/master/CONTRIBUTING.md) I run the command `python3 runtests.py` and I got an error, based on the python version that I use in my virtualenv vs the one listed in the ini file (3.7).

**To Reproduce**

```python
python3 runtests.py
```

**Expected Behavior**

No errors

**Actual Behavior**

```
run self: ['/Users/alessioizzo/alessio/mypy/venv/bin/python3', '-m', 'mypy', '--config-file', 'mypy_self_check.ini', '-p', 'mypy']
mypy/config_parser.py:16: error: Cannot find implementation or library stub for module named "tomli"  [import]
        import tomli as tomllib
    ^
mypy/config_parser.py:16: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
mypy/modulefinder.py:22: error: Cannot find implementation or library stub for module named "tomli"  [import]
        import tomli as tomllib
    ^
Found 2 errors in 2 files (checked 169 source files)

FAILED: self
```

**Your Environment**

- Mypy version used: latest from git clone
- Mypy command-line flags:
- Mypy configuration options from `mypy.ini` (and other config files): mypy_self_check.ini
- Python version used: 3.11
- OS: macos Apple M1 Pro, Ventura 13.1 

I guess that the solution can be to remove the python version from the ini file. I'm going to open a pull request for this one if this happens to be a bug.

### PR DESCRIPTION
**Pull request by aless10**
Fixes #14487 
(Explain how this PR changes mypy.)

This PR would like to explicit the fact that when you follow the steps in the [Contributing Guidelines](https://github.com/python/mypy/blob/master/CONTRIBUTING.md) and you run the command that runs the tests, the python version is `3.7` (set in the `mypy_self_check.ini` file).

[Open Source Saturday](https://www.meetup.com/it-IT/Open-Source-Saturday-Milano/)

### CODE REVIEW

File: mypy-requirements.txt [hunk 1]
```diff
@@ -2,4 +2,4 @@
 typing_extensions>=3.10
 mypy_extensions>=0.4.3
 typed_ast>=1.4.0,<2; python_version<'3.8'
-tomli>=1.1.0; python_version<'3.11'
+tomli>=1.1.0; python_version<'3.12'
```
Comment (JukkaL) [inline mypy-requirements.txt:5]: We only want to install tomli on Python 3.11 if we are running tests, so this should only be added to `test-requirements.txt` for Python 3.11 and later. If you change `mypy-requirements.txt`, tomli will always be installed, even though it's unnecessary for running mypy (it's only needed for self check and compilation).

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -50,6 +50,8 @@ like this:
 ```
 python3 runtests.py
 ```
+*Please note that the python version used here is ``3.7``. This is due to the fact that we want to always type check in 3.7 mode to make sure we don't use any features from more recent Python versions.*
+*You can change this in the ``mypy_self_check.ini`` file.*
```
Comment (JukkaL) [inline CONTRIBUTING.md:54]: I'm not sure if we should change `CONTRIBUTING.md`. If the dependencies are set up correctly, things should just work, and these additions are unnecessary detail. I don't think that changing the version is part of a normal workflow -- it should only be needed when dropping support for a Python version that has reached its end of life, and that's not something we need to document here.

File: test-requirements.txt [hunk 3]
```diff
@@ -17,3 +17,4 @@ pytest-cov>=2.10.0
 py>=1.5.2
 setuptools>=65.5.1
 six
+tomli>=1.1.0; python_version<'3.12'
```
Comment (JelleZijlstra) [inline test-requirements.txt:20]: Why only below 3.12?

**Review by JukkaL — APPROVED** [review]
Thanks for fixing this! It makes getting started developing smoother.

**Pull request merged**